src/nfts/saga.ts

```diff
@@ -112,17 +116,108 @@ export function* findCelebratedNft({ payload: { nfts } }: PayloadAction<FetchNft
     celebratedNftFound({
       networkId: celebratedNft.networkId,
       contractAddress: celebratedNft.contractAddress,
+      expirationDate: expirationDateString,
+      rewardReminderDate: rewardReminderDateString,
+      deepLink,
     })
   )
 }
 
+export function* findNftReward({ payload: { nfts } }: PayloadAction<FetchNftsCompletedAction>) {
+  const featureGateEnabled = getFeatureGate(StatsigFeatureGates.SHOW_NFT_CELEBRATION)
+  if (!featureGateEnabled) {
+    return
+  }
+
+  const {
+    celebratedNft,
+    expirationDate: expirationDateString,
+    rewardReminderDate: rewardReminderDateString,
+    deepLink,
+  } = getDynamicConfigParams(DynamicConfigs[StatsigDynamicConfigs.NFT_CELEBRATION_CONFIG])
+
+  if (
+    !celebratedNft ||
+    !celebratedNft.networkId ||
+    !celebratedNft.contractAddress ||
+    !expirationDateString ||
+    !rewardReminderDateString ||
+    !deepLink
+  ) {
+    return
+  }
+
+  const expirationDate = Date.parse(expirationDateString)
+  const rewardReminderDate = Date.parse(rewardReminderDateString)
+
+  if (Number.isNaN(expirationDate)) {
+    Logger.error(TAG, 'Invalid expiration date in remote config')
+    return
+  }
+
+  if (Number.isNaN(rewardReminderDate)) {
+    Logger.error(TAG, 'Invalid reminder date in remote config')
+    return
+  }
+
+  const expired = isPast(expirationDate)
+  if (expired) {
+    return
+  }
+
+  const userOwnsCelebratedNft = nfts.some(
+    (nft) => isSameNftContract(nft, celebratedNft) && !!nft.metadata
+  )
+  if (!userOwnsCelebratedNft) {
+    return
+  }
+
+  // TODO: check if the user is in allowed list (see ACT-1100)
+
+  const lastNftCelebration = yield* select(nftCelebrationSelector)
+  if (!lastNftCelebration) {
+    return
+  }
+
+  if (!isSameNftContract(lastNftCelebration, celebratedNft)) {
+    return
+  }
+
+  // sync values in the redux state with remote config
+  const valuesToSync = {
+    expirationDate: expirationDateString,
+    rewardReminderDate: rewardReminderDateString,
+    deepLink,
+  }
+
+  const showReminder = isToday(rewardReminderDate) || isPast(rewardReminderDate)
```

Comment:
is `isPast` supposed to be here?

Reply:
I believe, yes.

here we are checking if we are within the reminder period.

if user opens the app within the reminder period, but not exactly on the reminder start date, `rewardReminderDate` would be in the past.

for example:
* user opened app: March, 2 
* `rewardReminderDate`: March, 1 (`isPast === true`)

Reply:
OH of course, you're right. i was confusing this with the expiry date